### ISSUE TITLE
Methane Digester Engine does not output standard RF/energy cept to its own machines

### ISSUE THREAD

**Issue by Mementh** [GenDeathrow/Hatchery#144]
the generator/engine does not seem to output RF or any power to systems :(
using thermal expansion pipes fyi

**Comment by GenDeathrow** [comment]
Yeah they have worked with pretty much every RF capable devices I have ever tested them on. Also when you report issues to a Dev. Make sure you give MC versions and the version of hatchery your using cause I have no idea what to look for.

**Comment by GenDeathrow** [comment]
well technically that shouldn't be happening.. they don't care what side it pulls from. So that part is actually strange.

**Comment by joselitosn** [comment]
So I saw this bug report closed and I'm not sure if I should open another or use this one.

Anyways, I'm having the same behaviour using Thermal Dynamics fluxducts. No matter which side I connect, i can't transfer energy out of the Methane Digester Generator. The weird thing is that if I connect one charged Energy Cell and set it to output, the generator starts filling it's internal battery. I was only able to transfer energy out of the generator if I place the Energy Cell directly at any side of the generator.

Minecraft 1.12.2
Forge 14.23.2.2619
Thermal Dynamics 2.3.10.4
Hatchery 2.1.2

**Comment by joernott** [comment]
Having the same issues with Minecraft 1.12.2, Forge 1.12.2-14.23.3.2655, Hatchery 2.1.2 and xnet-1.6.9

**Comment by joernott** [comment]
I don't have the COFH mods in my pack, and I am using the latest xnet, so I will just go with updating Forge this evening

**Comment by Syndaryl** [comment]
If you're still having trouble, check if you're pulling from the sides, or the top/bottom - as mentioned upthread there have been issues with the top/bottom and that might have returned.

Actually a screenshot or two of your problematic layout would be useful for @GenDeathrow to replicate your issue. I'll add some screenshots of my config for people interested in CoFH when I get home.

**Comment by joselitosn** [comment]
Since people are having trouble to replicate this, here is a simple setup:

Mods installed:

Hatchery 2.1.2
Thermal Dynamics 2.4.1.10
Thermal Expansion 5.4.2.27

Place a Methane Digester Generator on the ground and an Energy Cell (Basic) one block away from any side. Then place a Leadstone Fluxduct between them. Make sure to config every side of the Energy Cell to input (blue).

![working](https://user-images.githubusercontent.com/422206/40210249-38fb3a7e-5a1a-11e8-89e9-f13b73ebcdb5.png)

This setup works fine. Now just config the Energy Cell receiving side to output (orange). It stops receiving energy as expected. Again fine.

![working-noinput](https://user-images.githubusercontent.com/422206/40210353-cef8adfe-5a1a-11e8-9ebc-1127f15aa197.png)

**NOW** connect the same fluxduct on any other receiving side of the Energy Cell.

![notworking](https://user-images.githubusercontent.com/422206/40210433-4f2dedd6-5a1b-11e8-900d-ca69623f8eaf.png)

No energy transfered.

Now let's connect disconnect from the receiving side and connect to any side on the Methane Digester Generator.

![notworking-feedback](https://user-images.githubusercontent.com/422206/40210537-d0dd96f6-5a1b-11e8-800b-53a18258a27f.png)

The generator receives the energy back from the Energy Cell?!?

I also noticed that even after the energy buffer of the Methane Digester Generator is full, it keeps emiting sounds and particles, like it is still working, although nothing is being consumed.

I'm not sure if @GenDeathrow is still following this issue, if not i will open one later.

**Comment by GenDeathrow** [comment]
I'll test this out. but there isn't any code to allow the generator to receive energy. So that should not happen. I use Forge energy technically, so unless COFH changed how they handle FE, I have no idea why its doing this.

**Comment by Natesky9** [comment]
Can confirm still an issue, generators will only output their rf from the two sides with the vents
